src/handlers/TechRecordsListHandler.ts

```diff
@@ -0,0 +1,165 @@
+    let techRecordItems: T[] = data.Items;
+    if (status !== STATUS.ALL) {
+      techRecordItems = this.filterTechRecordsByStatus(techRecordItems, status);
+    }
+    return techRecordItems;
+  }
+
+  public formatTechRecordItemForResponse(techRecordItem: T) {
+    // Adding primary and secondary VRMs in the same array
+    const vrms = [];
+    if (techRecordItem.primaryVrm) {
+      vrms.push({ vrm: techRecordItem.primaryVrm, isPrimary: true });
+    }
+    if (techRecordItem.secondaryVrms) {
+      for (const secondaryVrm of techRecordItem.secondaryVrms) {
+        vrms.push({ vrm: secondaryVrm, isPrimary: false });
+      }
+    }
+    Object.assign(techRecordItem, {
+      vrms
+    });
+    // Cleaning up unneeded properties
+    delete techRecordItem.primaryVrm; // No longer needed
+    delete techRecordItem.secondaryVrms; // No longer needed
+    delete techRecordItem.partialVin; // No longer needed
+
+    techRecordItem.techRecord.forEach((techRecord: any) => {
+      if (
+        techRecord.euroStandard !== undefined &&
+        techRecord.euroStandard !== null
+      ) {
+        techRecord.euroStandard = techRecord.euroStandard.toString();
+      }
+    });
+
+    return techRecordItem;
+  }
+  /* #endregion */
+
+  /* #region  Private functions */
+  private filterTechRecordsByStatus(techRecordItems: T[], status: string): T[] {
+    const recordsToReturn = [];
+    for (let techRecordItem of techRecordItems) {
+      techRecordItem = this.filterTechRecordsForIndividualVehicleByStatus(
+        techRecordItem,
+        status
+      );
+      if (techRecordItem.techRecord.length > 0) {
+        recordsToReturn.push(techRecordItem);
+      }
+    }
+    return recordsToReturn;
+  }
+
+  private filterTechRecordsForIndividualVehicleByStatus(
+    techRecordItem: T,
+    status: string
+  ): T {
+    const originalTechRecordItem = cloneDeep(techRecordItem);
+    let provisionalOverCurrent = false;
+    if (status === STATUS.PROVISIONAL_OVER_CURRENT) {
+      provisionalOverCurrent = true;
+      status = STATUS.PROVISIONAL;
+    }
+
+    techRecordItem.techRecord = techRecordItem.techRecord.filter(
+      (techRecord: any) => {
+        return techRecord.statusCode === status;
+      }
+    );
+
+    const { length } = originalTechRecordItem.techRecord;
+    const { statusCode } = originalTechRecordItem.techRecord[0];
+
+    if (
+      provisionalOverCurrent &&
+      length === 1 &&
+      techRecordItem.techRecord.length > 0 &&
+      (statusCode === STATUS.CURRENT || statusCode === STATUS.PROVISIONAL)
+    ) {
+      return techRecordItem;
+    }
+
+    if (
+      provisionalOverCurrent &&
+      (length === techRecordItem.techRecord.length ||
+        0 === techRecordItem.techRecord.length)
+    ) {
+      techRecordItem = this.filterTechRecordsForIndividualVehicleByStatus(
+        originalTechRecordItem,
+        STATUS.CURRENT
+      );
+    }
+
+    if (techRecordItem.techRecord.length <= 0) {
```

Comment:
We can move that at the top of the function I think

Reply:
This if condition is checking the outcome of the filters above this line of code.